### ISSUE TITLE
Loss masking for certain values such as out of bounds

### ISSUE THREAD

**Issue by surajpaib** [ganslate-team/ganslate#31]
Proposed solution: 

The loss functions must be modified to have the loss reduction to none so that per voxel loss is obtained. Once this is obtained, 
1. Create a mask based on matching with the certain value for the input image. 
2. Used masked_select to mask the loss to obtain a 1D tensor: https://pytorch.org/docs/stable/generated/torch.masked_select.html
3.  Reduce this 1D tensor obtaining an appropriate loss.

This loss masking needs to go into L1, SSIM and all the GAN losses.
